dataloader.py

```diff
@@ -142,7 +149,36 @@ def _one_hot_encode(self, image, mask):
         
         return image, one_hot_map
 
-    def data_batch(self, batch_size, augment, shuffle=False, one_hot_encode=False):
+    def _map_function(self, images_path, masks_path):
+        image, mask = self._parse_data(images_path, masks_path)
+
+        if self.augment:
+            if self.compose:
+                image, mask = self._corrupt_brightness(image, mask)
+                image, mask = self._corrupt_contrast(image, mask)
+                image, mask = self._corrupt_saturation(image, mask)
+                image, mask = self._crop_random(image, mask)
+                image, mask = self._flip_left_right(image, mask)
+            else:
+                options = [self._corrupt_brightness,
+                           self._corrupt_contrast,
+                           self._corrupt_saturation,
+                           self._crop_random,
+                           self._flip_left_right]
+                augment_func = random.choice(options)
```

Comment:
It seems TF records the choice the first time the sampling happens (if random crop was sampled, then it is always applied since the graph of the pipeline is constructed on the first line). We probably need add tf.cond variables to apply random transforms for each image

Reply:
It looks like TF compiles the map function to a static graph if that is the case then the first selected augmentation function will be always used. To solved this scenario I have decorated the map function with ```tf.function``` which knows how to deal with py functions like random and compile it. 
I tried several times and it works, please do your tests and merge :)